### DISCUSSION TITLE
Terminate Gröbner basis computation after a given time

### DISCUSSION THREAD

**Discussion by oskarhenriksson** [oscar-system/Oscar.jl#4218] [Q&A]
Is there a way to automatically end a long-running Gröbner basis computation in Oscar after a certain amount of time (say, 60 seconds), without killing the kernel? There doesn't seem to be any standard timeout functionality in Julia, and when I try to use tricks like [this](https://discourse.julialang.org/t/simple-timeout-of-function/99578/4) for `groebner_basis`, the computation doesn't actually end after the given time frame.

**Comment by fingolfin** [comment]
@hannes14 do we have such a functionality?

**Comment by hannes14** [comment]
Singular has Singular.Libgeneral.watchdog (https://www.singular.uni-kl.de/Manual/4-4/sing_1044.htm),
but it forks a new process and takes its data as strings
